validphys2/src/validphys/core.py

```diff
@@ -515,28 +528,55 @@ def __str__(self):
         return self.name
 
 class FKTableSpec(TupleComp):
-    def __init__(self, fkpath, cfactors):
+    """
+    In Legacy Mode each fktable has one single path.
+    For pineappl tables instead a FKTable is formed by any number of grids
+    therefore in order to check whether we have a new-format or old-format table
+    we will just check whether fkpath is a list
+    For now holds the metadata as an attribute to this function.
+    This is useless/transitional since this metadata is already in the new CommonData format
+    """
+    def __init__(self, fkpath, cfactors, metadata=None):
+        self.cfactors = cfactors if cfactors is not None else []
+        self.legacy = True
+        if isinstance(fkpath, (tuple, list)):
+            self.legacy = False
+            fkpath = tuple(fkpath)
         self.fkpath = fkpath
-        self.cfactors = cfactors
-        super().__init__(fkpath, cfactors)
+        self.metadata = metadata
 
-    #NOTE: We cannot do this because Fkset owns the fktable, and trying
-    #to reuse the loaded one fails after it gets deleted.
-    #@functools.lru_cache()
-    def load(self):
+        # If this is a yaml file that loads an applgrid, keep also the name of the target
+        # this is needed since we can now easily reutilize grids 
+        if not self.legacy and self.metadata.get("appl"):
+            super().__init__(fkpath, cfactors, self.metadata.get("target_dataset"))
+        else:
+            super().__init__(fkpath, cfactors)
+
+
+    def _load_legacy(self):
         return FKTable(str(self.fkpath), [str(factor) for factor in self.cfactors])
 
+    def _load_pineappl(self):
+        log.info("Reading: %s", self.fkpath)
+        return pineappl_reader(self)
+
+    def load_with_cuts(self, cuts):
+        """Load the fktable and apply cuts inmediately. Returns a FKTableData"""
+        return load_fktable(self).with_cuts(cuts)
+
+    def load(self):
+        if self.legacy:
+            return self._load_legacy()
+        return self._load_pineappl()
+
+
 class PositivitySetSpec(DataSetSpec):
     """Extends DataSetSpec to work around the particularities of the positivity datasets"""
 
     def __init__(self, name, commondataspec, fkspec, maxlambda, thspec):
         cuts = Cuts(commondataspec, None)
         self.maxlambda = maxlambda
         super().__init__(name=name, commondata=commondataspec, fkspecs=fkspec, thspec=thspec, cuts=cuts)
-        if len(self.fkspecs) > 1:
-            # The signature of the function does not accept operations either
-            # so more than one fktable cannot be utilized
-            raise ValueError("Positivity datasets can only contain one fktable")
```

Comment:
Because the positivity fktables in `libNNPDF` were read _almost_ in the same way as the other fktables but not quite. The change to vp meant that one could read wrong positivity tables (and it would've work silently) so this is just so vp was crashing as well.

After this PR the positivity fktables are read in the same way that other fktables so this "legacy crash" is not needed anymore.